src/GlobalTouchModel.js

```diff
@@ -0,0 +1,163 @@
+(function ()
+{
+	'use strict';
+
+	function bind(fn, ctx)
+	{
+		return function ()
+		{
+			return fn.apply(ctx, arguments);
+		};
+	}
+
+	function tryCall(fn, eventObject)
+	{
+		try
+		{
+			fn(eventObject);
+		}
+		catch (e)
+		{
+			console.error(e);
+		}
+	}
+
+	var TouchModel = {
+		// Public Methods
+		/**
+		 * Registers an element for touch events
+		 * @param {HMTLElement} element
+		 * @returns {TouchDelegate} Returns the TouchDelegate for an element
+		 */
+		register: function register(element)
+		{
+			if (!this.active)
+				this._attachToDocument();
+			if (!element.touchDelegate)
+				element.touchDelegate = new touchDelegate();
+
+			return element.touchDelegate;
+		},
+		/**
+		 * Deregisters an element for touch events, releasing the reference to the touchDelegate
+		 * @param {HTMLElement} element
+		 * @returns undefined
+		 */
+		deregister: function deregister(element)
+		{
+			if (element.touchDelegate)
+				element.touchDelegate = null;
+		},
+		// Private Variables
+		_active: false,
+
+		_handles:
+		{
+			touchstart: bind(function start(e)
+			{
+				var element = e.target,
+					event;
+
+				this._attachListener('touchmove');
+				this._attachListener('touchend');
+
+				while (element.parentNode)
+				{
+					if (element.touchDelegate)
+					{
+						if (!event)
+							event = this._createEvent('touchstart', e);
+						element.touchDelegate.call('touchstart', event);
+					}
+					element = element.parentNode;
+				}
+			}, this),
+
+			touchmove: bind(function move(e)
+			{
+				var element = e.target,
+					event;
+
+				while (element.parentNode)
+				{
+					if (element.touchDelegate)
+					{
+						if (!event)
+							event = this._createEvent('touchmove', e);
+						element.touchDelegate.call('touchmove', event);
+					}
+					element = element.parentNode;
+				}
+			}, this),
```

Comment:
Context used in binding is probably wrong here (during declaration not execution so likely window instead of TouchModel)